src/_pytest/fixtures.py

```diff
@@ -1124,13 +1124,25 @@ def getfixtureinfo(self, node, func, cls, funcargs=True):
             argnames = getfuncargnames(func, cls=cls)
         else:
             argnames = ()
+
+        # Collecting parametrize arguments so they are ignored later
+        parametrize_argnames = []
```

Comment:
Could you move this code to its own method to improve readability? This is quite complicated as it is, it would be nice to make it as easy to understand as possible. Also please add a docstring with your comment and more details if you would, including the issue that resulted in this bug. 👍

Reply:
Done